hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -53,3 +56,10 @@ def test_is_IP6_addr(address):
     assert all(len(part) == 4 for part in as_hex)
     raw = unhexlify(address.replace(u":", u"").encode("ascii"))
     assert len(raw) == 16
+
+
+def test_invalid_domain_arguments():
+    with pytest.raises(InvalidArgument):
+        domains(max_element_length=64).example()
+    with pytest.raises(InvalidArgument):
```

Comment:
Worth using a `match=` to assert on the contents of the error message?

Reply:
Potentially, but given that this is explicit invocation of a know out of bounds condition to test for rejection in the  validation logic before returning any results and the constraints wont cause this exception to come from any of the inner strategy logic... I'm undecided if this is a good enhancement or being overly specific. I wouldn't mind some extra opinions on this.

Reply:
IMO it's fine as-is; we don't usually test for the details of argument validation messages.  I *would* go with Alex's suggestions above though to standardise the messages.

Reply:
Its probably also worth adding lower bound/negative input assertion checks to properly test both ends of validity checking logic.

Reply:
At that point I usually reach for `@pytest.mark.parametrize("kwds", [dict(...), ...])` so I can just list out invalid cases.